### ISSUE TITLE
Implement `MAX_SOUND_COUNT_IN_MEMORY` check in `Sampler::addSound`

### ISSUE THREAD

**Issue by izzyreal** [izzyreal/mpc#246]
**Current situation**
`Sampler::addSound(int)` and `Sampler::addSound()` (which delegates to the former) do not check if the maximum number of sounds in memory (256 for the MPC2000XL) has already been reached or not.

**Desired situation**
Said methods should perform said check. Maybe they could return an empty `std::shared_ptr` if the maximum has been reached, but maybe another mechanism is better. Give it some more thought.

Either way, clients need to deal with a rejected request. How they do this exactly, should be checked on the original case by case, and then replicated.

In code, we can find the following clients of `addSound()`:
```
Usages in All Places  (19 usages found)
    Member access  (19 usages found)
        vmpc2000xl  (19 usages found)
            editables/mpc/src/main/disk  (2 usages found)
                ApsLoader.cpp  (1 usage found)
                    loadSound  (1 usage found)
                        263 auto sound = mpc.getSampler()->addSound();
                ProgramLoader.cpp  (1 usage found)
                    loadProgram  (1 usage found)
                        112 auto sound = mpc.getSampler()->addSound();
            editables/mpc/src/main/lcdgui/screens  (2 usages found)
                LoadScreen.cpp  (1 usage found)
                    loadSound  (1 usage found)
                        465 auto sound = sampler->addSound();
                SampleScreen.cpp  (1 usage found)
                    function  (1 usage found)
                        122 auto sound = sampler->addSound();
            editables/mpc/src/main/lcdgui/screens/dialog  (1 usage found)
                ResampleScreen.cpp  (1 usage found)
                    function  (1 usage found)
                        86 auto destSnd = sampler->addSound();
            editables/mpc/src/main/lcdgui/screens/window  (1 usage found)
                EditSoundScreen.cpp  (1 usage found)
                    function  (1 usage found)
                        416 auto newSample = sampler->addSound();
            src/main/gui  (1 usage found)  <--------------------- vmpc-juce
                PadControl.cpp  (1 usage found)
                    loadFile  (1 usage found)
                        77 auto sound = mpc.getSampler()->addSound();
```
and of `addSound(int)`:
```
Usages in All Places  (9 usages found)
    Member access  (7 usages found)
        vmpc2000xl  (7 usages found)
            editables/mpc/src/main  (1 usage found)
                AutoSave.cpp  (1 usage found)
                    restoreAutoSavedState  (1 usage found)
                        104 auto sound = mpc.getSampler()->addSound(sndReader.getSampleRate());
            editables/mpc/src/main/lcdgui/screens/dialog  (3 usages found)
                MonoToStereoScreen.cpp  (1 usage found)
                    function  (1 usage found)
                        135 auto newSound = sampler->addSound(left->getSampleRate());
                StereoToMonoScreen.cpp  (2 usages found)
                    function  (2 usages found)
                        93 auto left = sampler->addSound(sound->getSampleRate());
                        94 auto right = sampler->addSound(sound->getSampleRate());
            editables/mpc/src/main/lcdgui/screens/window  (2 usages found)
                EditSoundScreen.cpp  (2 usages found)
                    function  (2 usages found)
                        629 auto newSample = sampler->addSound(sound->getSampleRate());
                        646 auto newSample = sampler->addSound(sound->getSampleRate());
            src/main  (1 usage found) <----------------------------------- vmpc-juce
                VmpcProcessor.cpp  (1 usage found)
                    setStateInformation  (1 usage found)
                        635 auto sound = mpc.getSampler()->addSound(sndReader.getSampleRate());
```

**Comment by izzyreal** [comment]
* In LOAD screen load an APS: The problem does not exist. APS flushes sounds in memory before loading the ones from the APS file.

* In LOAD screen load a PGM: It loads sounds until sound directory is full. It displays the error popup over the "Load a Program" window. Any button push takes user back to LOAD screen.

  There's some strange behaviour, but it's not important to emulate. For completion, a quick description. There's 255 sounds in memory. We want to load a program with 3 sounds: KICK1, SNARE4 and HAT1 (listed in that order in the PGM file). First round of loading, only KICK1 is loaded. Now there's 256 sounds in memory. When we try to load the PGM again, KICK1 is replaced by SNARE4. When we try to load again, SNARE4 is replaced by KICK1.

* In LOAD screen load individual SND/WAV: Error popup is displayed on top of LOAD screen. Any button push takes user back to LOAD screen.

* SAMPLE screen: The user is blocked from arming the recorder, i.e. pressing F6 yields the error popup on top of the SAMPLE screen. Button push takes user back to SAMPLE screen.

**Comment by izzyreal** [comment]
Resolved in 6b7f3df1b222e5062f0e8a32adb115be3e3a006b and izzyreal/vmpc-juce@349a226869b3dc5b84999f57823ad56be4ce79fd
